### PR TITLE
Implemented pthread_barrier_functions

### LINKED ISSUE #1645: Extend pthread compatibility
We need to implement following functions:
```
int pthread_barrier_destroy(pthread_barrier_t );
int pthread_barrier_init(pthread_barrier_t restrict, const pthread_barrierattr_t restrict, unsigned);
int pthread_barrier_wait(pthread_barrier_t );
int pthread_barrierattr_destroy(pthread_barrierattr_t *);
```
This changes are required for #1643

### PR DESCRIPTION
**Pull request by sobhaniiest**
Fixes #1645

### CODE REVIEW

**Comment by sobhaniiest** [comment]
There are some mistakes in code style. I'll do another commit along with any further changes suggested by you after your reviews. Thanks :)

**Comment by Deryugin** [comment]
Thanks for your contribution!

I'm trying to build this test, but it gives me compilation error

```
src/tests/posix/pthread/pthread_barrier_test.c:26:24: error: initialization of ‘pthread_barrier_t’ {aka ‘struct _pthread_barrier_t *’} from incompatible pointer type ‘struct _pthread_barrier_t **’ [-Werror=incompatible-pointer-types]
   26 |  pthread_barrier_t b = (pthread_barrier_t *) arg;
      |        
```

Fix it, please

**Comment by sobhaniiest** [comment]
Hi @Deryugin. I have fixed the error and have updated a test and the code style. Please review.
Thanks :)

**Comment by sobhaniiest** [comment]
The CI failed for the  TEMPLATE=usermode86/debug and showed
ci_error

I have not changed anything regarding this. Am I doing something wrong?

**Comment by sobhaniiest** [comment]
But in the forked repo, it's showing all checks have passed, please have a check [here](https://github.com/sobhaniiest/embox/runs/506290096).

File: src/tests/posix/pthread/pthread_barrier_test.c [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+#include <embox/test.h>
+#include <unistd.h>
+#include <stdlib.h>
+
+EMBOX_TEST_SUITE("posix/pthread_barrier_api");
+
+TEST_CASE ("Initialize barrier with pthread_barrier_init()") {
+	pthread_barrier_t barrier = NULL; 
+	pthread_barrierattr_t attr = (pthread_barrierattr_t) malloc(sizeof(*attr));
+	attr->pshared = PTHREAD_PROCESS_PRIVATE;
```
Comment (Deryugin) [inline src/tests/posix/pthread/pthread_barrier_test.c]: As I can see from the standard (https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_types.h.html), `pthread_barrierattr_t` is not explicitly required to be a pointer, so test should not rely on fact that it is a pointer, i.e. `sizeof(*attr)` may fail to compile if implementation will change (but will still be POSIX-compliant).

Also it means that you can't access `attr->pshared` directly.

Comment (sobhaniiest) [reply src/tests/posix/pthread/pthread_barrier_test.c]: Okay! [According to this](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_types.h.html), `pthread_barrier_t` also should not be a pointer type.  Right?

Comment (Deryugin) [reply src/tests/posix/pthread/pthread_barrier_test.c]: It's unspecified, so it's implementation-dependent.

**Pull request closed without merging**